### PR TITLE
[SDPA-5606] Add published log to admin_audit_trail

### PR DESCRIPTION
**Pull request by mayurngondhkar**
### Jira
https://digital-engagement.atlassian.net/browse/SDPA-5606

### Change
Add Published log to admin audit trail

Testing
PR: https://nginx-php.pr-1247.content-vic.sdp2.sdp.vic.gov.au/admin/reports/audit-trail

### CODE REVIEW

File: tide_core.module [hunk 1]
```diff
@@ -448,3 +448,27 @@ function tide_core_system_breadcrumb_alter(Breadcrumb &$breadcrumb, RouteMatchIn
...
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function tide_core_node_presave(NodeInterface $node) {
+  $node_state = $node->get('moderation_state')->getValue()[0]['value'];
+
+  if ($node_state === 'published') {
+    $log = [
+      'type' => 'node',
+      'operation' => 'published',
+      'description' => t('%type: %title - New node created with workflow state %new_state', [
```
Comment (anthony-malkoun) [inline tide_core.module:462]: Is it necessarily a new node though? I would check this language with the product owner as they don't really talk in terms of nodes.

**Comment by mayurngondhkar** [comment]
Testing
PR: https://nginx-php.pr-1247.content-vic.sdp2.sdp.vic.gov.au/admin/reports/audit-trail

The log is updated once a user publishes a node.

**Review by anthony-malkoun — CHANGES_REQUESTED** [review]
Can you just check the language of the message please.

File: tide_core.module (hunk 1 above)
Comment (mayurngondhkar) [reply tide_core.module:462]: @anthony-malkoun Will check with the product owner.

File: tide_core.module [hunk 2]
```diff
@@ -448,3 +448,27 @@ function tide_core_system_breadcrumb_alter(Breadcrumb &$breadcrumb, RouteMatchIn
...
+        '%type' => $node->getType(),
+        '%title' => $node->getTitle(),
+        '%new_state' => $node_state,
+      ]),
+      'ref_numeric' => $node->id(),
+      'ref_char' => $node->getTitle(),
+    ];
+
+    // Add the log to the "admin_audit_trail" table
```
Comment (MdNadimHossain) [inline tide_core.module:471]: add a full stop at the end of the comment else it will throw lint issue.

File: tide_core.module [hunk 3]
```diff
@@ -448,3 +448,27 @@ function tide_core_system_breadcrumb_alter(Breadcrumb &$breadcrumb, RouteMatchIn
...
+    admin_audit_trail_insert($log);
```
Comment (MdNadimHossain) [inline tide_core.module:472]: Although Audit trail is a dependency of tide_core, but this is a function getting called from another module, I would put a check for function_exist for a better protection. eg 
```
if(function_exists('admin_audit_trail_insert')) {
  admin_audit_trail_insert($log);
}
```

**Review by MdNadimHossain — CHANGES_REQUESTED** [review]
Just couple of things.

File: tide_core.module (hunk 2 above)
Comment (mayurngondhkar) [reply tide_core.module]: Fixed

File: tide_core.module (hunk 1 above)
Comment (mayurngondhkar) [reply tide_core.module:463]: Hey @anthony-malkoun,

Checked with Nick. About the term 'node'.

He requested the use of the term 'page' for better readability.

Changes are made to reflect this.

File: tide_core.module (hunk 3 above)
Comment (mayurngondhkar) [reply tide_core.module]: Change made

**Review by anthony-malkoun — APPROVED** [review]
Thanks for making changes Mayur.

**Pull request merged**